lib/core/presentation/widgets/bottom_bar/bottom_bar_widget.dart

```diff
@@ -0,0 +1,182 @@
+import 'dart:io';
+
+import 'package:app/core/application/auth/auth_bloc.dart';
+import 'package:app/core/presentation/widgets/bottom_bar/app_tabs.dart';
+import 'package:app/core/presentation/widgets/lemon_circle_avatar_widget.dart';
+import 'package:app/core/presentation/widgets/loading_widget.dart';
+import 'package:app/core/utils/avatar_utils.dart';
+import 'package:app/theme/color.dart';
+import 'package:auto_route/auto_route.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+import 'package:flutter_vibrate/flutter_vibrate.dart';
+
+class BottomBar extends StatefulWidget {
+  @override
+  _BottomBarState createState() => _BottomBarState();
+}
+
+class _BottomBarState extends State<BottomBar>
+    with SingleTickerProviderStateMixin {
+  AppTab _selectedTab = AppTab.home;
+  late AnimationController _animationController;
+  late Animation<double> _animation;
+  bool _isTabChanged = false;
```

Comment:
Declare those variable as `final`

Reply:
actually you don't need to override `initState` to declare these 2. You can put it here.
For example:
```
late final _animationController = AnimationController(
      vsync: this,
      duration: const Duration(milliseconds: 450),
    );
```

```
late Animation<double> _animation = Tween<double>(begin: 0, end: 1).animate(
      CurvedAnimation(
        parent: _animationController,
        curve: Curves.easeInOut,
      ),
    );
```

Reply:
Since they need access to the vsync and other contextual information, it's not possible to directly initialize them as final fields at the class level.